src/iter.rs

```diff
@@ -669,8 +668,12 @@ where
     pub fn iter_all_chunks_in_bounds(
         bound_min: L,
         bound_max: L,
-        max_depth: u64,
+        max_depth: u8,
     ) -> ChunksInBoundIter<L> {
+        debug_assert!(
```

Comment:
Not sure if I like this.

For me it would make more sense that it simply returns no chunks if the bounds are empty